### ISSUE TITLE
Wallaby is flagging an imported variable as undefined.

### ISSUE THREAD

**Issue by martinduo** [wallabyjs/public#2559]
### Issue description or question

This is the same issue I reported earlier (#2556), but in different code. As before I run npx jest and all tests pass. I import a variable that has been loaded with a pointer to a child node of my main planObject. Wallaby thinks the variable is undefined. The earlier issue you requested the files involved, so I'm providing them now. The problem is at line 145 in file _moduleDataSubRoutines.test.js where the variable 'insurances' is referenced. That variable is initialized at line 70 in file moduleObjectRoutines.js by a call from line 133 in file _moduleDataSubRoutines.test.js.

Here is a link to a Dropbox folder with the files. I don't know how useful they will be. There are so many dependencies. Let me know if you need more.

https://www.dropbox.com/sh/tumustk2xhbem1g/AADXJ3OUlKOERb3B2Jvj_SUNa?dl=0

### Wallaby diagnostics report

```
{
  editorVersion: '1.50.1',
  pluginVersion: '1.0.248',
  editorType: 'VSCode',
  osVersion: 'win32 10.0.19041',
  nodeVersion: 'v12.18.3',
  coreVersion: '1.0.984',
  checksum: 'NWE2NTY4MGE2MDgzYTY2NDFhOWM5M2NmZmQ4OGQyYzYsMTYwNTgzMDQwMDAwMCwx',
  config: {
    diagnostics: {
      jest: {
        config: {
          configs: [
            {
              automock: false,
              browser: false,
              cache: true,
              cacheDirectory: 'C:\\Users\\Bruce\\AppData\\Local\\Temp\\jest',
              clearMocks: false,
              coveragePathIgnorePatterns: [ '\\\\node_modules\\\\' ],
              cwd: 'C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend',
              dependencyExtractor: undefined,
              detectLeaks: undefined,
              detectOpenHandles: undefined,
              displayName: undefined,
              errorOnDeprecated: false,
              extraGlobals: [],
              filter: undefined,
              forceCoverageMatch: [],
              globalSetup: undefined,
              globalTeardown: undefined,
              globals: {},
              haste: { computeSha1: false, providesModuleNodeModules: [], throwOnModuleCollision: false },
              moduleDirectories: [ 'node_modules' ],
              moduleFileExtensions: [ 'js', 'json', 'jsx', 'ts', 'tsx', 'node' ],
              moduleLoader: undefined,
              moduleNameMapper: [],
              modulePathIgnorePatterns: [],
              modulePaths: undefined,
              name: '8cc8ef3a1a03669e0e821e7c83daf783',
              prettierPath: 'prettier',
              resetMocks: false,
              resetModules: false,
              resolver: undefined,
              restoreMocks: false,
              rootDir: 'C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend',
              roots: [ 'C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend' ],
              runner: 'jest-runner',
              setupFiles: [],
              setupFilesAfterEnv: [],
              skipFilter: false,
              skipNodeResolution: undefined,
              snapshotResolver: undefined,
              snapshotSerializers: [],
              testEnvironment: 'C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend\\node_modules\\jest-environment-jsdom\\build\\index.js',
              testEnvironmentOptions: {},
              testLocationInResults: false,
              testMatch: [ '**/__tests__/**/*.[jt]s?(x)', '**/?(*.)+(spec|test).[tj]s?(x)' ],
              testPathIgnorePatterns: [
                'C:\\\\Users\\\\Bruce\\\\Dropbox\\\\Code\\\\NODE BACKEND\\\\MoneyPlanBackend\\\\dist\\\\',
                'C:\\\\Users\\\\Bruce\\\\Dropbox\\\\Code\\\\NODE BACKEND\\\\MoneyPlanBackend\\\\node_modules\\\\'
              ],
              testRegex: [],
              testRunner: 'C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend\\node_modules\\jest-jasmine2\\build\\index.js',
              testURL: 'http://localhost',
              timers: 'real',
              transform: [ [ '.(js|jsx|ts|tsx)', 'C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend\\node_modules\\@sucrase\\jest-plugin\\dist\\index.js', {} ] ],
              transformIgnorePatterns: [ '\\\\node_modules\\\\' ],
              unmockedModulePathPatterns: undefined,
              watchPathIgnorePatterns: []
            }
          ],
          globalConfig: {
            bail: 0,
            changedFilesWithAncestor: false,
            changedSince: undefined,
            collectCoverage: false,
            collectCoverageFrom: [ 'src/**/*.js', '!src/**/*.debug.js', '!**/node_modules/**' ],
            collectCoverageOnlyFrom: undefined,
            coverageDirectory: 'C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend\\coverage',
            coverageProvider: 'babel',
            coverageReporters: [ 'json', 'text', 'lcov', 'clover' ],
            coverageThreshold: undefined,
            detectLeaks: undefined,
            detectOpenHandles: undefined,
            enabledTestsMap: undefined,
            errorOnDeprecated: false,
            expand: false,
            filter: undefined,
            findRelatedTests: false,
            forceExit: false,
            globalSetup: undefined,
            globalTeardown: undefined,
            json: false,
            lastCommit: false,
            listTests: undefined,
            logHeapUsage: false,
            maxConcurrency: 5,
            maxWorkers: 7,
            noSCM: undefined,
            noStackTrace: false,
            nonFlagArgs: undefined,
            notify: false,
            notifyMode: 'failure-change',
            onlyChanged: false,
            onlyFailures: false,
            outputFile: undefined,
            passWithNoTests: undefined,
            projects: [],
            replname: undefined,
            reporters: undefined,
            rootDir: 'C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend',
            runTestsByPath: false,
            silent: undefined,
            skipFilter: false,
            testFailureExitCode: 1,
            testNamePattern: undefined,
            testPathPattern: '',
            testResultsProcessor: undefined,
            testSequencer: 'C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend\\node_modules\\@jest\\test-sequencer\\build\\index.js',
            testTimeout: undefined,
            updateSnapshot: 'new',
            useStderr: false,
            verbose: undefined,
            watch: false,
            watchAll: false,
            watchPlugins: undefined,
            watchman: true
          },
          hasDeprecationWarnings: false,
          wallaby: {
            roots: [],
            watchPathIgnorePatterns: [ '\\\\node_modules\\\\', '\\.\\\\dist\\\\|\\.\\\\build\\\\|\\.\\\\coverage\\\\|\\\\\\..+\\\\', '\\\\dist\\\\' ],
            testPathIgnorePatterns: [ '\\\\node_modules\\\\', '\\.\\\\dist\\\\|\\.\\\\build\\\\|\\.\\\\coverage\\\\|\\\\\\..+\\\\', '\\\\dist\\\\' ],
            testMatch: [ '**/__tests__/**/*.[jt]s?(x)', '**/?(*.)+(spec|test).[tj]s?(x)' ],
            testRegex: []
          }
        }
      }
    },
    testFramework: { version: 'jest@24.8.0', configurator: 'jest@24.8.0', reporter: 'jest@24.8.0', starter: 'jest@24.8.0', autoDetected: true },
    filesWithCoverageCalculated: [ 'src/**/*.js', '!src/**/*.debug.js', '!**/node_modules/**' ],
    globalSetup: false,
    micromatch: true,
    files: [
      { pattern: '\\\\node_modules\\\\', regexp: /\\node_modules\\/, ignore: true, trigger: true, load: true },
      { pattern: '\\.\\\\dist\\\\|\\.\\\\build\\\\|\\.\\\\coverage\\\\|\\\\\\..+\\\\', regexp: /\.\\dist\\|\.\\build\\|\.\\coverage\\|\\\..+\\/, ignore: true, trigger: true, load: true },
      { pattern: '\\\\dist\\\\', regexp: /\\dist\\/, ignore: true, trigger: true, load: true },
      { pattern: '**/**', ignore: false, trigger: true, load: true, order: 1 },
      { pattern: '**/__tests__/**/*.[jt]s?(x)', ignore: true, trigger: true, load: true },
      { pattern: '**/?(*.)+(spec|test).[tj]s?(x)', ignore: true, trigger: true, load: true }
    ],
    tests: [
      { pattern: '\\\\node_modules\\\\', regexp: /\\node_modules\\/, ignore: true, trigger: true, load: true, test: true },
      {
        pattern: '\\.\\\\dist\\\\|\\.\\\\build\\\\|\\.\\\\coverage\\\\|\\\\\\..+\\\\',
        regexp: /\.\\dist\\|\.\\build\\|\.\\coverage\\|\\\..+\\/,
        ignore: true,
        trigger: true,
        load: true,
        test: true
      },
      { pattern: '\\\\dist\\\\', regexp: /\\dist\\/, ignore: true, trigger: true, load: true, test: true },
      { pattern: '**/__tests__/**/*.[jt]s?(x)', ignore: false, trigger: true, load: true, test: true, order: 2 },
      { pattern: '**/?(*.)+(spec|test).[tj]s?(x)', ignore: false, trigger: true, load: true, test: true, order: 3 }
    ],
    filesWithNoCoverageCalculated: [],
    runAllTestsInAffectedTestFile: false,
    updateNoMoreThanOneSnapshotPerTestFileRun: false,
    compilers: {},
    preprocessors: {},
    maxConsoleMessagesPerTest: 100,
    autoConsoleLog: true,
    delays: { run: 0, edit: 100, update: 0 },
    workers: { initial: 0, regular: 0, recycle: false },
    teardown: undefined,
    hints: {
      ignoreCoverage: '__REGEXP /ignore coverage|istanbul ignore/',
      ignoreCoverageForFile: '__REGEXP /ignore file coverage/',
      commentAutoLog: '?',
      testFileSelection: { include: '__REGEXP /file\\.only/', exclude: '__REGEXP /file\\.skip/' }
    },
    automaticTestFileSelection: true,
    runSelectedTestsOnly: false,
    extensions: {},
    env: { type: 'node', params: {}, runner: 'C:\\Program Files\\nodejs\\node.exe', viewportSize: { width: 800, height: 600 }, options: { width: 800, height: 600 }, bundle: true },
    reportUnhandledPromises: true,
    slowTestThreshold: 75,
    lowCoverageThreshold: 80,
    loose: true,
    configCode: 'auto.detect#575196754'
  },
  packageJSON: {
    dependencies: { 'base-64': '^0.1.0', 'big.js': '^5.2.2', dotenv: '^8.2.0', 'lodash.clonedeep': '^4.5.0', seedrandom: '^3.0.5', uuid: '^8.3.0' },
    devDependencies: {
      '@sucrase/jest-plugin': '^2.0.0',
      '@types/jest': '^25.1.4',
      'babel-eslint': '^10.1.0',
      eslint: '^6.8.0',
      'eslint-config-airbnb-base': '^14.0.0',
      'eslint-config-prettier': '^6.10.0',
      'eslint-plugin-import': '^2.20.1',
      'eslint-plugin-jest-formatting': '^1.2.0',
      'eslint-plugin-node': '^11.0.0',
      'eslint-plugin-promise': '^4.2.1',
      jest: '^25.1.0',
      nodemon: '^2.0.2',
      prettier: '^2.0.5',
      sucrase: '^3.12.1',
      xml2js: '^0.4.23'
    }
  },
  fs: { numberOfFiles: 106 },
  debug: [
    '2020-11-07T19:29:16.992Z angular/cli config Angular CLI not found.\n',
    '2020-11-07T19:29:17.115Z jest/config Detected Jest.\n',
    '2020-11-07T19:29:17.115Z jest/config Configured Jest.\n',
    '2020-11-07T19:29:17.116Z project Wallaby Node version: v12.18.3\n',
    '2020-11-07T19:29:17.116Z project Wallaby config: C:\\Users\\Bruce\\Dropbox\\Code\\NODE BACKEND\\MoneyPlanBackend\\auto.detect\n',
    '2020-11-07T19:29:18.127Z project File cache: C:\\Users\\Bruce\\.vscode\\extensions\\wallabyjs.wallaby-vscode-1.0.248\\projects\\d3e4866eedb79af9\n',
    '2020-11-07T19:29:18.150Z uiService Listening port 52742\n',
    '2020-11-07T19:29:18.176Z workers Parallelism for initial run: 6, for regular run: 3\n',
    '2020-11-07T19:29:18.176Z workers Starting run worker instance #0\n',
    '2020-11-07T19:29:18.176Z workers Starting run worker instance #1\n',
    '2020-11-07T19:29:18.176Z workers Starting run worker instance #2\n',
    '2020-11-07T19:29:18.176Z workers Starting run worker instance #3\n',
    '2020-11-07T19:29:18.176Z workers Starting run worker instance #4\n',
    '2020-11-07T19:29:18.176Z workers Starting run worker instance #5\n',
    '2020-11-07T19:29:18.177Z workers Web server is listening at 55765\n',
    '2020-11-07T19:29:18.210Z project File cache requires some updates, waiting required files from IDE\n',
    '2020-11-07T19:29:18.260Z project Stopping process pool\n',
    '2020-11-07T19:29:18.261Z project Test run started; run priority: 3\n',
    '2020-11-07T19:29:18.262Z project Running all tests\n',
    '2020-11-07T19:29:18.264Z workers Starting test run, priority: 3\n',
    '2020-11-07T19:29:18.265Z workers Distributing tests between 6 workers\n',
    '2020-11-07T19:29:18.265Z workers Running tests in parallel\n',
    '2020-11-07T19:29:18.266Z nodeRunner Starting sandbox [worker #0, session #5a966]\n',
    '2020-11-07T19:29:18.266Z nodeRunner Starting sandbox [worker #1, session #t5nyd]\n',
    '2020-11-07T19:29:18.266Z nodeRunner Starting sandbox [worker #2, session #tqqq6]\n',
    '2020-11-07T19:29:18.266Z nodeRunner Starting sandbox [worker #3, session #ko91l]\n',
    '2020-11-07T19:29:18.266Z nodeRunner Starting sandbox [worker #4, session #5ocks]\n',
    '2020-11-07T19:29:18.266Z nodeRunner Starting sandbox [worker #5, session #yq4pt]\n',
    '2020-11-07T19:29:18.266Z nodeRunner Preparing sandbox [worker #0, session #5a966]\n',
    '2020-11-07T19:29:18.266Z nodeRunner Preparing sandbox [worker #1, session #t5nyd]\n',
    '2020-11-07T19:29:18.266Z nodeRunner Preparing sandbox [worker #2, session #tqqq6]\n',
    '2020-11-07T19:29:18.266Z nodeRunner Preparing sandbox [worker #3, session #ko91l]\n',
    '2020-11-07T19:29:18.266Z nodeRunner Preparing sandbox [worker #4, session #5ocks]\n',
    '2020-11-07T19:29:18.266Z nodeRunner Preparing sandbox [worker #5, session #yq4pt]\n',
    '2020-11-07T19:29:18.330Z workers Started run worker instance (delayed) #0\n',
    '2020-11-07T19:29:18.330Z nodeRunner Prepared sandbox [worker #0, session #5a966]\n',
    '2020-11-07T19:29:18.330Z workers [worker #0, session #5a966] Running tests in sandbox\n',
    '2020-11-07T19:29:18.337Z workers Started run worker instance (delayed) #1\n',
    '2020-11-07T19:29:18.338Z nodeRunner Prepared sandbox [worker #1, session #t5nyd]\n',
    '2020-11-07T19:29:18.338Z workers [worker #1, session #t5nyd] Running tests in sandbox\n',
    '2020-11-07T19:29:18.338Z workers Started run worker instance (delayed) #2\n',
    '2020-11-07T19:29:18.338Z nodeRunner Prepared sandbox [worker #2, session #tqqq6]\n',
    '2020-11-07T19:29:18.338Z workers [worker #2, session #tqqq6] Running tests in sandbox\n',
    '2020-11-07T19:29:18.342Z workers Started run worker instance (delayed) #3\n',
    '2020-11-07T19:29:18.342Z nodeRunner Prepared sandbox [worker #3, session #ko91l]\n',
    '2020-11-07T19:29:18.342Z workers [worker #3, session #ko91l] Running tests in sandbox\n',
    '2020-11-07T19:29:18.349Z workers Started run worker instance (delayed) #4\n',
    '2020-11-07T19:29:18.349Z nodeRunner Prepared sandbox [worker #4, session #5ocks]\n',
    '2020-11-07T19:29:18.349Z workers [worker #4, session #5ocks] Running tests in sandbox\n',
    '2020-11-07T19:29:18.349Z workers Started run worker instance (delayed) #5\n',
    '2020-11-07T19:29:18.350Z nodeRunner Prepared sandbox [worker #5, session #yq4pt]\n',
    '2020-11-07T19:29:18.350Z workers [worker #5, session #yq4pt] Running tests in sandbox\n',
    '2020-11-07T19:29:21.196Z workers [5ocks] Loaded unknown number of test(s)\n',
    '2020-11-07T19:29:21.210Z workers [5ocks] Test executed: Handling Entries\n',
    '2020-11-07T19:29:21.259Z workers [5ocks] Run 1 test(s), skipped 0 test(s)\n',
    '2020-11-07T19:29:21.260Z workers [5ocks] Sandbox is responsive, closing it\n',
    '2020-11-07T19:29:21.284Z workers [tqqq6] Loaded unknown number of test(s)\n',
    '2020-11-07T19:29:21.284Z workers [tqqq6] Test executed: Handling Items\n',
    '2020-11-07T19:29:21.298Z workers [tqqq6] Loaded unknown number of test(s)\n',
    '2020-11-07T19:29:21.298Z workers [tqqq6] Test executed: should throw Error with correct name and message\n',
    '2020-11-07T19:29:21.298Z workers [tqqq6] Test executed: should throw Error with correct name and message\n',
    '2020-11-07T19:29:21.299Z workers [tqqq6] Test executed: should throw Error with correct name and message\n',
    '2020-11-07T19:29:21.308Z workers [tqqq6] Run 4 test(s), skipped 0 test(s)\n',
    '2020-11-07T19:29:21.310Z workers [tqqq6] Sandbox is responsive, closing it\n',
    '2020-11-07T19:29:21.614Z workers [yq4pt] Loaded unknown number of test(s)\n',
    '2020-11-07T19:29:21.614Z workers [yq4pt] Test executed: formats percent values into expected strings\n',
    '2020-11-07T19:29:21.615Z workers [yq4pt] Test executed: formats dollar values into expected string\n',
    '2020-11-07T19:29:21.616Z workers [yq4pt] Test executed: formats person birthdate\n',
    '2020-11-07T19:29:21.618Z workers [yq4pt] Test executed: clears All Persons TaxData \n',
    '2020-11-07T19:29:21.623Z workers [yq4pt] Test executed: counts Sources Of PayoutFunds \n',
    '2020-11-07T19:29:21.623Z workers [yq4pt] Test executed: tests some PlanEndYear considerations\n',
    '2020-11-07T19:29:21.624Z workers [yq4pt] Test executed: tests person life considerations\n',
    '2020-11-07T19:29:21.625Z workers [yq4pt] Test executed: tests plan fund balances \n',
    '2020-11-07T19:29:21.625Z workers [yq4pt] Test executed: tests for missing names, dups, joint owners\n',
    '2020-11-07T19:29:21.625Z workers [yq4pt] Test executed: formats person birthdate\n',
    '2020-11-07T19:29:21.626Z workers [yq4pt] Test executed: checks for correct insured persons\n',
    '2020-11-07T19:29:21.642Z workers [yq4pt] Run 11 test(s), skipped 0 test(s)\n',
    '2020-11-07T19:29:21.643Z workers [yq4pt] Sandbox is responsive, closing it\n',
    '2020-11-07T19:29:21.677Z workers [ko91l] Loaded unknown number of test(s)\n',
    '2020-11-07T19:29:21.677Z workers [ko91l] Test executed: Handling Entries\n',
    '2020-11-07T19:29:21.678Z workers [ko91l] Loaded unknown number of test(s)\n',
    '2020-11-07T19:29:21.678Z workers [ko91l] Test executed: can access imported accounts reference\n',
    '2020-11-07T19:29:21.679Z workers [ko91l] Test executed: can add breadcrumbs and collection methods\n',
    '2020-11-07T19:29:21.679Z workers [ko91l] Test executed: should iterate all accounts\n',
    '2020-11-07T19:29:21.679Z workers [ko91l] Test executed: should get the array at parentArrayList \n',
    '2020-11-07T19:29:21.681Z workers [ko91l] Test executed: should get the collection object at parentPlanObject \n',
    '2020-11-07T19:29:21.701Z workers [ko91l] Run 6 test(s), skipped 0 test(s)\n',
    '2020-11-07T19:29:21.702Z workers [ko91l] Sandbox is responsive, closing it\n',
    '2020-11-07T19:29:21.727Z workers [t5nyd] Loaded unknown number of test(s)\n',
    '2020-11-07T19:29:21.728Z workers [t5nyd] Test executed: Handling items\n',
    '2020-11-07T19:29:21.729Z workers [t5nyd] Loaded unknown number of test(s)\n',
    '2020-11-07T19:29:21.729Z workers [t5nyd] Test executed: tests the availablity of payout sources \n',
    '2020-11-07T19:29:21.747Z workers [t5nyd] Run 2 test(s), skipped 0 test(s)\n',
    '2020-11-07T19:29:21.748Z workers [t5nyd] Sandbox is responsive, closing it\n',
    '2020-11-07T19:29:21.822Z workers [5a966] Loaded unknown number of test(s)\n',
    '2020-11-07T19:29:21.822Z workers [5a966] Test executed: Handling items\n',
    '2020-11-07T19:29:21.823Z workers [5a966] Loaded unknown number of test(s)\n',
    '2020-11-07T19:29:21.823Z workers [5a966] Test executed: gets the named example planObject\n',
    '2020-11-07T19:29:21.823Z workers [5a966] Test executed: moves specified parameter change\n',
    '2020-11-07T19:29:21.824Z workers [5a966] Test executed: tests all assets to move any specified parameter change\n',
    '2020-11-07T19:29:21.824Z workers [5a966] Test executed: adds a 2nd parameter change for Bobby College Payments\n',
    '2020-11-07T19:29:21.824Z workers [5a966] Test executed: adds a 3rd parameter change for Bobby College Payments\n',
    '2020-11-07T19:29:21.825Z workers [5a966] Test executed: adds a 2nd parameter change for global parameters\n',
    '2020-11-07T19:29:21.825Z workers [5a966] Test executed: confirms existing parameter changes\n',
    '2020-11-07T19:29:21.825Z workers [5a966] Test executed: consolidates ParamChangeDictionary of asset\n',
    '2020-11-07T19:29:21.844Z workers [5a966] Run 9 test(s), skipped 0 test(s)\n',
    '2020-11-07T19:29:21.846Z workers [5a966] Sandbox is responsive, closing it\n',
    '2020-11-07T19:29:21.846Z workers Merging parallel test run results\n',
    '2020-11-07T19:29:21.856Z project Test run finished\n',
    '2020-11-07T19:29:21.856Z project Processed console.log entries\n',
    '2020-11-07T19:29:21.856Z project Processed loading sequences\n',
    '2020-11-07T19:29:21.857Z project Test name duplicate: should throw Error with correct name and message\n',
    '2020-11-07T19:29:21.857Z project Test name duplicate: should throw Error with correct name and message\n',
    '2020-11-07T19:29:21.857Z project Processed executed tests\n',
    '2020-11-07T19:29:21.880Z project Processed code coverage\n',
    '2020-11-07T19:29:22.187Z project Test run result processed and sent to IDE\n'
  ]
}
```

**Comment by smcenlly** [comment]
Thanks for providing the updated diagnostic report and your application files.

Unfortunately we couldn't simply run the files that you provided because of the missing dependencies but we have a theory as to what is going on.

We can see from your code sample that when you import your variable (`insurances`), the value is initially undefined. The `createPlanObjectReferences` that is called in your `beforeAll()` method is then setting the internals of `moduleObjectRoutines.js` which updates the imported variable's value. We think that your `beforeAll()` function is failing when it's being run in Wallaby for some reason. Jest default behavior is for test execution to continue even if `beforeAll()` or `beforeEach()` errors for some reason. For erxample:

![image](https://user-images.githubusercontent.com/2074089/98487688-21949800-2270-11eb-9286-72bed58780ca.png)

Unfortunately, without being able to reproduce your error, we're guessing at what exactly is wrong. Are you able to provide us with the minimal set of code files so we can reproduce your issue? If you are happy to share your code (just not in a public forum), you may email it to use: hello@wallabyjs.com.

**Comment by martinduo** [comment]
Thanks for your theory. I will work on setting up something you can run
Jest and Wallaby on to create the problem.

**Comment by martinduo** [comment]
Very strange...

This morning I started to create a smaller set of files to demo the
problem. First I copied a full backup (except for the node_modules) of my
repo from an external HD. Then after running npm install, I started
Wallaby. And the insurances //? line showed the object, no problem!

Thinking that the npm install fixed it, I then copied my live repo to
another folded and started Wallaby.  And the insurances //? line showed the
object, no problem! This was a straight file to file copy of the whole repo
including the node_module folder. I didn't change a thing (knowingly) and
it works!

Back at the source repo, the insurances //? line in
_moduleDataSubRoutines.test.js still shows undefined. So I proceed to
comment out most of the imports. And I comment out the whole first
describe() block. the line still shows insurances //? undefined.

Now I go to the modulesObjectRoutines file to prune stuff there. I comment
out the first import line, and bam, the insurances //? line back in
_moduleDataSubRoutines.test.js shows the expected object. When I undo all
changes in all files, the insurances //? line still shows the expected
object.

Finally I reload VSCode, and when it starts (Wallaby is still running), the
insurances //? line still shows the expected object. No matter what I do
now: reload VSCode, start/stop Wallaby, the insurances //? line shows the
expected object.

It seems like just doing a small update in the file that was exporting the
insurances variable caused the file to be evaluated again and the
insurances variable with the assigned value was available in the importing
file. So maybe when the exporting file recieved the insurances value, it
was blocked from evaluating? This is my first node project and my first
unit testing with Jest, so I'm in the dark about the plumbing.

Since npx Jest ran fine, the bug only blocks my use of Wallaby. But maybe
all I need to do is tweek the exporting file to unblock it.

Very frustrating that I can't provide you a demo of the bug.

**Comment by smcenlly** [comment]
Thanks for providing the additional detail. It sounds like perhaps there was an issue with Wallaby's cache of transformed files. We have spotted a potential conflict between `sucrase` (which your jest configuration is using to transform your code files) and Wallaby. We've updated the latest version of Wallaby to be more aggressive in invalidating the cache. Please the latest version of Wallaby over the next few days (version: `v1.0.986`) and you should not see the problem recur.

There are a couple of other things worth noting:

1. We were not previously familiar with [`sucrase`](https://github.com/alangpierce/sucrase) but after reviewing the package for jest testing, we would recommend use `babel-jest` instead. `sucrase` [does not support hoisting jest mocks](https://github.com/alangpierce/sucrase/issues/539), which would be a deal-breaker for us in terms of how we test using jest. If you didn't specifically intend on using `sucrase` instead of `babel`, then it may be worth re-considering your choice of transformation technology and switching to use babel.

1. The way that you are exporting values from your module (`moduleObjectRoutines`) and then mutating them is a bit of an anti-pattern that may lead to fragile code. It's not "wrong" but generally something we would try to avoid as it's an easy way to introduce unexpected behavior. We generally try to make our imports immutable where we can. In your case, you could return the values you need from `createPlanObjectReferences` and use those in your code instead of mutating the exported values. Again, not something that's inherently "wrong" but may be worth considering if you were not aware.

## Simplified version of current code
**moduleObjectRoutines.js**
```javascript
let insurances;

function createPlanObjectReferences(planObject) {
  insurances = planObject.insurances;
}

export { createPlanObjectReferences, insurances }
```

**_moduleDataSubRoutines.test.js**
```javascript
import { insurances } from './moduleObjectRoutines'
import { createPlanObjectReferences } from './moduleObjectRoutines'

describe('With Insurance2ndToDieTrust/moduleDataSubRoutines', () => {
  beforeAll(() => {
    createPlanObjectReferences({insurances:['123']});
  })

  it('checks for correct insured persons', () => {
    insurances //?

    // other test code...
  })
})
```

## Simplified version of non-mutating code
**moduleObjectRoutines.js**
```javascript
function createPlanObjectReferences(planObject) {
  const insurances = planObject.insurances;
  return { insurances }
}

export { createPlanObjectReferences, insurances }
```

**_moduleDataSubRoutines.test.js**
```javascript
import { createPlanObjectReferences } from './moduleObjectRoutines'

describe('With Insurance2ndToDieTrust/moduleDataSubRoutines', () => {
  let fixture;

  beforeAll(() => {
    fixture = createPlanObjectReferences({insurances:['123']});
  })

  it('checks for correct insured persons', () => {
    fixture.insurances //?

    // other test code...
  })
})
```

**Comment by martinduo** [comment]
Thanks for your answer. I've updated to .986. Thanks for your notes of
interest. That's kind of you to do that. Working alone, nobody sees my code.

I used sucrase to allow me to easily use es6 imports and exports while
waiting for node v14 to be LTS ( which just happened). I understand that
node14 supports es6 without the need for any special flags or babel
handling. So I'll soon upgrade from v12 and test it out.

My code has an interesting history. It is 33,000 lines of js code converted
from a legacy VB.Net app of mine that was programmed in Visual Studio were
every public variable was globally accessible in the project name space.
All those names were converted to exports and imports to connect the 47
files. All throughout the program references are made to these exported
names expecting them to be defined.

So I use the method shown in your "Simplified version of current code"
specifically to avoid converting every call to those names into the form of
fixture.name. The fixture name itself would be different depending on the
origin file.

I really appreciate your taking the time to write up those notes. Thanks.
